### ISSUE TITLE
New Profile Page

### ISSUE THREAD

**Issue by wintermeyer** [wintermeyer/vutuv#576]
# The Task

**Create a mockup profile HTML page.** English or German (language doesn't matter for this mockup). To solve this issue no Elixir code has to be written. Just plain old HTML and CSS (probably embedded in the HTML). If you need data use dummy data. Please store the file(s) in `misc/mockups/` or a subdirectory of it. The HTML file should contain the form. Later on someone will convert this mockup into a Phoenix template. But that will be a separate issue. 

Here is an example of the v1.1 version of that profile page: https://www.vutuv.de/users/stefan.wintermeyer

Please create a pull request to submit your work (master branch).

## New Features

Version 2.0 will offer features which will have to become part of the profile page:

- `Points`. A user gets points for specific tasks. Think https://stackoverflow.com reputations but for vutuv stuff.
- `Badges`. Depending on their point status users get specific badges.
- `Posts`. Think of it as a Blogpost. For new users it will be limited to 140 characters. Once a user gets through a specific level of points he/she can post 280 characters and 512 characters. As a bonus users can use Markdown once they achieved a specific point level. See #577 
- `Votes`. Users can up and down vote posts (like stack overflow.com)
- Share. Like a retweet in Twitter.
- `Message`. Think DM of Twitter. I can send a message to everybody who follows me and vice versa.

# Alert Area

It would be very useful to have an area where system messages or warnings can be displayed.

# General Thoughts about the Profile Page

The biggest problem is that we don't have the resources to build custom made CSS for all the sub resources (e.g. https://www.vutuv.de/users/stefan.wintermeyer/work_experiences or https://www.vutuv.de/users/stefan.wintermeyer/links). So we need some sort of generic approach.

I think (but please tell me if this is a wrong assumption) that we need to stand on some frameworks shoulder (Bootstrap comes to mind) so that Phoenix developer can create sub resource pages by themself.

Bottom line: We need one nice looking profile page with all the information and a generic set for all the sub resource pages.

@MarioRicalde and @xamiam: Please share your thoughts about this.

**Comment by MarioRicalde** [comment]
All these sub-resource pages are really needed? Are they showing some information that is not visible in other places? I'm asking because:

https://www.vutuv.de/users/stefan.wintermeyer/work_experiences/freelancer.wintermeyer.consulting

Doesn't really show anything new that is not visible on a previous page.

**Comment by wintermeyer** [comment]
> All these sub-resource pages are really needed? 

Very good question. I have never thought about this this way. 

The only reason for this view is that sometimes there are too many elements to be shown on the profile page. If somebody has a long work history or 20 urls you can’t show them all on the profile and would therefore link to the sub resource index page. 

I’m open for alternatives.

**Comment by MarioRicalde** [comment]
Realistically speaking you want to plan for the 99% of the users and not for the 1% of the users. If someone goes above and beyond to have so many, then we should just show them and keep a super long page. Because from what I understand, that's what they want anyway. Your average Joe is not going to be putting 100 job history places, just the bigger ones that make them look good. I reckon.

There's also a concern with SEO ( if this is even crawlable ) with a lot of the content coming up as pretty much duplicate content.

**Comment by wintermeyer** [comment]
> Realistically speaking you want to plan for the 99% of the users and not for the 1% of the users. If someone goes above and beyond to have so many, then we should just show them and keep a super long page.

I see your point. Your right. No more sub resource pages for show information.

**Comment by wintermeyer** [comment]
But what are we going to do with the posts? Everyone who’s using that new feature will accumulate more post than being possible to display on the profile page.

**Comment by MarioRicalde** [comment]
Posts are understandable a thing which we should HOPE to get a lot of. Those are paginated, and they should have their own child design. These, however, shouldn't be too generic and look like everything else, needs to be special. I think.

**Comment by MarioRicalde** [comment]
@wintermeyer I'm jumping to work on this one, I have a couple of questions about the implementation of this particular section. Do we intend to keep that idea of inline styling? Or here we're using normal CSS, will we have access to JS here?

**Comment by wintermeyer** [comment]
> Do we intend to keep that idea of inline styling?

I don't know yet. Currently most visitors of vutuv.de come via Google or via direct link to a profile. Speed is key for all those potential new users. They don't have the CSS in their local cache yet. So for them I'm thinking: Let's try to inline it.

For already existing users (who would have a cookie set) a separate CSS file makes more sense because their browser can cache it.

Bottom line: Let's inline it for the version 2.0 and let's tackle the proper - if has cookie - rule in version 2.1.

> will we have access to JS here?

That is a tricky one too and we have the same situation as with the above question. 

For users who already have a vutuv account and the set cookie JavaScript is fine (but not an over the board amount of JavaScript because it does slow down the WebPerformance). For all potential new users who don't have a cookie: I'm thinking that they don't need JavaScript. They can not interact with the page anyway without having an account. So a non JavaScript page would be super fast for them.

BUT: I'm open for other solutions for both questions. Speed is just super important for vutuv. Right now we can not compete feature wise with LinkedIn. But we can compete speed wise and for many - especially mobile - users this is a big plus and I often get positiv feedback about vutuv's speed.

**Comment by MarioRicalde** [comment]
Understood, I have one extra question:

Do we intend to let users customize the looks of their profile? Think Twitter cover image or Facebook cover image.

**Comment by wintermeyer** [comment]
> Do we intend to let users customize the looks of their profile? Think Twitter cover image or Facebook cover image.

Not in v2.0. We have enough other stuff on our hands.

**Comment by wintermeyer** [comment]
We need to have some sort of “Member since mm/yy” text on the profile.

**Comment by MarioRicalde** [comment]
@wintermeyer roger that.

I'm struggling a bit trying to figure out what we want to show as the centerpiece of the design. My understanding is that we want to focus on showing information about the User Experience. But in the old design, we have Followers & Following as somewhat of a centerpiece. 

Could you provide me with a list of things in order of importance for that page? So that I can figure out how to arrange it in a clean way. That'd help me a lot.

**Comment by wintermeyer** [comment]
The problem: When people use the post feature then those posts are important. But there will be plenty of people who don’t post and use it just to be found by human resource departments. 

In v1 we didn’t have posts. Followers seemed to be the most interesting thing to display because it is a business network. 

Long story short: I see your point but I don’t have the one-size-fits-all answer.

**Comment by wintermeyer** [comment]
hmmm... if I don’t know you but for some reason land on your profile page what would help me to figure out if I like/need you?

- The tags and their ratings (e.g. CSS or Origami)
- How many points you have. On stackoverflow.com one always checks how many points an account has. 
- Your posts. 
- Your avatar. We are humans.

**Comment by MarioRicalde** [comment]
Help me understand the Points functionality, what gives you points? What happens when you click points, do you go to a view where you show what give you what points?

**Comment by wintermeyer** [comment]
A user gets points for interactions. 

- He logs in: 1 point
- She writes a post: 5 points
- The post gets shared by other users: 2 points per share
- etc

Points can be redeemed (e.g. buying ads). But points age. They lose 20% of their value every 12 months. 

A specific amount of points grants special privileges. E.g. being able to write longer posts.

**Comment by riverrun** [comment]
What needs to be done with this issue? Can we close it?
